linux/tests/sync_linux_ut/sync_linux_ut.c

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdatomic>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdatomic.h>
+#endif
+
+#include <limits.h>
+
+#include <sys/syscall.h>
+#include <linux/futex.h>
+#include <time.h>
+
+
+#include "azure_macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+#include "umock_c/umock_c.h"
+
+#include "sync.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+#define ENABLE_MOCKS
+#include "mock_sync.h"
+#undef ENABLE_MOCKS
+
+
+static bool check_timeout;
+static uint32_t expected_timeout_ms;
+static int expected_return_val;
+static int hook_mock_syscall(long call_code, int* uaddr, int futex_op, int val, const struct timespec* timeout, int* uaddr2, int val3)
+{
+    /*Tests_SRS_SYNC_LINUX_43_001: [ wait_on_address shall initialize a timespec struct with .tv_nsec equal to timeout_ms* 10^6. ]*/
+    if(check_timeout)
+    {
+        ASSERT_ARE_EQUAL(long, expected_timeout_ms*1e6, timeout->tv_nsec);

```

Comment:
>tv_nsec [](start = 65, length = 7)

Isn't there another field you are setting on this struct? It should also be checked.